sdk/ml/azure-ai-ml/azure/ai/ml/identity/_credentials/_AzureMLSparkOnBehalfOfCredential.py

```diff
@@ -56,6 +56,7 @@ def _get_client_args(**kwargs) -> Optional[dict]:
 
     token_service_endpoint = os.environ.get("AZUREML_SYNAPSE_TOKEN_SERVICE_ENDPOINT")
     obo_access_token = os.environ.get("AZUREML_OBO_CANARY_TOKEN")
+    obo_endpoint = os.environ.get("AZUREML_OBO_USER_TOKEN_FOR_SPARK_RETRIEVAL_API") or "getuseraccesstokenforspark"
```

Comment:
You can use `get(key, value)` for default value if key is not found

Reply:
updated